ls1intum/artemis-ios
# `General`: Release 1.3.0

## New
- Filters for Conversations
- Support for Profile Pictures
- Simpler titles for exercise, lecture & exam channels
- Split View Navigation on iPad
- Group exercises & lectures by Past, Due Soon, Current and Future instead of by week
- Redesign for Course Cards on the Dashboard

## Fixed
- Toolbar for editing a message was editing the wrong text field
- When deleting the first message in a group of messages, the header would disappear
- Removed buttons for adding/removing users from course-wide channels
- Messages disappearing after being sent in non course wide channels 

## Changed
- When no exercise/lecture/exam channels exist, the corresponding category is not shown either

## Testing
### How to install
1. <a href="https://testflight.apple.com/join/canwoVCk">Join Artemis TestFlight Beta: testflight.apple.com/join/canwoVCk</a>
2. Download & Install Latest Build of Version 1.3

### What to test
- Verify that all described bugs have been fixed
- Ensure that navigating to exercises, lectures and conversations still works
- Filters for conversations (All/Unread/Favorites), Note: Filters only show up if there are any matches for that category 
- Check whether profile pictures are shown
- Make sure sending and replying to messages etc still works

You can use a test server by entering its url on the login screen under "Select university">"Custom instance".

### Known issues
- If you are at the very bottom of a chat and long press on a message near the top, some actions might be off screen. Workaround: Scroll up by at least 1px before long pressing.
- You cannot swipe to go back from the Course View to the Dashboard due to an iOS 18 workaround.
